src/core/src/energy/functions.rs

```diff
@@ -487,4 +531,26 @@ mod tests {
         let e1 = born.energy(4.0 + EPS);
         assert_relative_eq!((e0 - e1) / EPS, born.force(4.0), epsilon=1e-6);
     }
+    
+  #[test]
+    fn morse() {
+        let morse = MorsePotential{width: 2.0, r0: 1.3, depth: 4.0};
+        
+        //http://www.wolframalpha.com/input/?i=4+*+(1+-+e%5E(-2+*+(1+-+1.3)))%5E2
+        assert_eq!(morse.energy(1.0), 2.703517287822119);
+        //http://www.wolframalpha.com/input/?i=4+*+(1+-+e%5E(-2+*+(-1+-+1.3)))%5E2
+        assert_eq!(morse.energy(-1.0), 38796.641709840165);
```

Comment:
That's an interesting test case 😄 There shouldn't be negative distances. But I guess it tests the function.